File: app/src/main/java/sgtmelon/scriptum/element/common/SingleDialog.java
package sgtmelon.scriptum.element.common;

import android.app.Dialog;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AlertDialog;
import sgtmelon.scriptum.R;
import sgtmelon.scriptum.office.annot.def.DialogDef;
import sgtmelon.scriptum.office.blank.DialogBlank;

public final class SingleDialog extends DialogBlank {

    private String[] name;
    private int init, check;

    public void setArguments(int check) {
        Bundle arg = new Bundle();

        arg.putInt(DialogDef.INIT, check);
        arg.putInt(DialogDef.VALUE, check);

        setArguments(arg);
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        Bundle arg = getArguments();

        if (savedInstanceState != null) {
            init = savedInstanceState.getInt(DialogDef.INIT);
            check = savedInstanceState.getInt(DialogDef.VALUE);
        } else if (arg != null) {
            init = arg.getInt(DialogDef.INIT);
            check = arg.getInt(DialogDef.VALUE);
        }

        return new AlertDialog.Builder(context)
                .setTitle(title)
                .setSingleChoiceItems(name, check, (dialogInterface, i) -> {
                    check = i;
                    setEnable();
                })
                .setPositiveButton(getString(R.string.dialog_btn_accept), onPositiveClick)
                .setNegativeButton(getString(R.string.dialog_btn_cancel), (dialog, id) -> dialog.cancel())
                .setCancelable(true)
                .create();
    }

    @Override
    public void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);

        outState.putInt(DialogDef.INIT, init);
        outState.putInt(DialogDef.VALUE, check);
    }

    public String[] getRows() {
        return name;
    }

    public void setRows(String[] name) {
        this.name = name;
    }

    public int getCheck() {
        return check;
    }

    @Override
    protected void setEnable() {
        super.setEnable();

        if (init == check) buttonPositive.setEnabled(false);
        else buttonPositive.setEnabled(true);
    }

}

File: safedialog/src/main/java/sgtmelon/safedialog/SingleDialog.kt
package sgtmelon.safedialog

import android.app.Dialog
import android.content.Context
import android.content.DialogInterface
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import sgtmelon.safedialog.annotation.NdValue
import sgtmelon.safedialog.annotation.SavedTag

/**
 * Dialog for single choice
 */
class SingleDialog : BlankDialog() {

    var itemArray: Array<String> = arrayOf()

    /**
     * If value is TRUE - it means what user can choose any dialog item. Otherwise see [setEnable].
     */
    var applyEnable: Boolean = false

    private var checkInit = NdValue.CHECK
    var check = NdValue.CHECK
        private set

    var itemListener: DialogInterface.OnClickListener? = null

    /**
     * Call before [show].
     */
    fun setArguments(check: Int) = apply {
        arguments = Bundle().apply {
            putInt(SavedTag.INIT, check)
            putInt(SavedTag.VALUE, check)
        }
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        super.onCreateDialog(savedInstanceState)

        return AlertDialog.Builder(context as Context)
            .setTitle(title)
            .setSingleChoiceItems(itemArray, check) { _, i ->
                itemListener?.onClick(dialog, i)
                check = i
                setEnable()
            }
            .setPositiveButton(getString(R.string.dialog_button_apply), onPositiveClick)
            .setNegativeButton(getString(R.string.dialog_button_cancel)) { dialog, _ -> dialog.cancel() }
            .setCancelable(true)
            .create()
            .applyAnimation()
    }

    override fun onRestoreContentState(savedInstanceState: Bundle) {
        super.onRestoreContentState(savedInstanceState)
        itemArray = savedInstanceState.getStringArray(SavedTag.LIST) ?: arrayOf()
    }

    override fun onRestoreInstanceState(bundle: Bundle?) {
        super.onRestoreInstanceState(bundle)
        checkInit = bundle?.getInt(SavedTag.INIT) ?: NdValue.CHECK
        check = bundle?.getInt(SavedTag.VALUE) ?: NdValue.CHECK
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putStringArray(SavedTag.LIST, itemArray)

        outState.putInt(SavedTag.INIT, checkInit)
        outState.putInt(SavedTag.VALUE, check)
    }

    override fun setEnable() {
        super.setEnable()
        positiveButton?.isEnabled = applyEnable || checkInit != check
    }
}